website/content/contribute/module/index.md

```diff
@@ -0,0 +1,199 @@
+4. [Register the Module](#register-the-module) to make it available to the control plane.
+
+These steps are described in the following sections in more detail, so that you can create your own modules for use by the Mesh for Data.  Note that a new module is maintained in its own git repository, separate from the core [Mesh for Data repository] (https://github.ibm.com/data-mesh-research/datamesh).
+
+
+## Module Workload
+
+The module workload is associated with a specific user workload and is deployed by the control plane.  It may implement the logic required itself, or it may be a client interface to an external component.  
+
+In either case its input parameters must match the relevant type of `ModuleArguments`.  The details of the different types of arguments are defined in [blueprint_types.go](/Users/sima/Work/Dev/go/sima-m4d/the-mesh-for-data/manager/apis/app/v1alpha1/blueprint_types.go). 
+
+```
+type ModuleArguments struct {
+	// CopyArgs are parameters specific to modules that do implicit copies of data from one data store to another.
+	// +optional
+	Copy *CopyModuleArgs `json:"copy,omitempty"`
+
+	// ReadArgs are parameters that are specific to modules that enable user workloads to read data or are involved in preparing data to be read.
+	// +optional
+	Read []ReadModuleArgs `json:"read,omitempty"`
+
+	// WriteArgs are parameters that are specific to modules that enable an application to write data
+	// +optional
+	Write []WriteModuleArgs `json:"write,omitempty"`
+}
+```
+
+The status of the module workload is monitored by the control plane via standard kubernetes mechanisms.  
+TODO: Expand on this or delete it?
+
+If the module workload needs to return information to the user, that information should be written to the rel.Info.Note. 
+TODO: Where should it be written to??
+
+## Module Helm Chart
+For any module chosen by the control plane to be part of the data path, the control plane needs to be able to install/remove/upgrade an instance of the module. Mesh for Data uses [Helm](https://helm.sh/docs/intro/using_helm/)
+to provide this functionality.  Thus, a [helm chart](https://helm.sh/docs/chart_template_guide/getting_started/) should be provided for each `Module Workload`.
+
+Example: [Arrow Flight Module helm chart](https://github.com/IBM/the-mesh-for-data-flight-module/tree/cd168bb6cdf666c2ec1df960395c0dc1c8feeaa9/helm/afm)
+
+
+## M4DModule YAML
+M4DModule is a kubernetes Custom Resource Definition (CRD) which describes to the control plane the functionality provided by the module.  The M4DModule CRD has no controller. The specification of the `M4DModule` Kubernetes CRD is available in the [API documentation]({{< baseurl >}}/docs/reference/api/generated/app/#k8s-api-github-com-ibm-the-mesh-for-data-manager-apis-app-v1alpha1-m4dmodule). 
+
+### Metadata about the module
+The first section in the YAML file contains general information about the module and where it will run.
```

Comment:
I suggest to remove any reference to the namespace. The admin will apply the module in m4d-system, so no need to mention it here.